utility/common_utils.cpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file common_utils.cpp
+ * @author Sam Olds (sam@ardusat.com)
+ * @date 4-8-2016
+ * @brief Common I2C read/write functions used by many drivers
+ */
+
+#include "common_utils.h"
+
+/**
+ * Generic I2C Read function
+ *
+ * @param devAddr I2C Device address
+ * @param reg Register address
+ * @param val pointer to byte array to read into
+ * @param length number of bytes to read
+ *
+ * @return 0 on success, other on failure
+ */
+int readFromRegAddr(uint8_t devAddr, uint8_t reg, void *val, int length, endian_e endianness)
```

Comment:
technically, `size_t` is probably more appropriate for the length parameter, see if that makes problems for you in any of the uses of this function though...